ResearchKit/Common/ORKResultPredicate.h

```diff
@@ -508,20 +838,48 @@ ORK_CLASS_AVAILABLE
  
  @param taskIdentifier                  The identifier of the task whose result you want to match.
                                             Pass `nil` to match the ongoing task.
+ @param stepIdentifier                  The identifier of the step whose result you want to match.
+                                            Passing `nil` assumes the step identifier is the same as
+                                            the result identifier.
  @param resultIdentifier                The identifier of the question result you are interested in.
- @param minimumExpectedAnswerHour       The minimum expected hour component value.
- @param minimumExpectedAnswerMinute     The minimum expected minute component value.
- @param maximumExpectedAnswerHour       The maximum integer hour component value.
- @param maximumExpectedAnswerMinute     The maximum expected minute component value.
+ @param minimumExpectedHour             The minimum expected hour component value.
+ @param minimumExpectedMinute           The minimum expected minute component value.
+ @param maximumExpectedHour             The maximum integer hour component value.
+ @param maximumExpectedMinute           The maximum expected minute component value.
  
  @return A result predicate.
  */
 + (NSPredicate *)predicateForTimeOfDayQuestionResultWithTaskIdentifier:(nullable NSString *)taskIdentifier
+                                                        stepIdentifier:(nullable NSString *)stepIdentifier
                                                       resultIdentifier:(NSString *)resultIdentifier
-                                             minimumExpectedAnswerHour:(NSInteger)minimumExpectedAnswerHour
-                                           minimumExpectedAnswerMinute:(NSInteger)minimumExpectedAnswerMinute
-                                             maximumExpectedAnswerHour:(NSInteger)maximumExpectedAnswerHour
-                                           maximumExpectedAnswerMinute:(NSInteger)maximumExpectedAnswerMinute;
+                                                   minimumExpectedHour:(NSInteger)minimumExpectedHour
+                                                 minimumExpectedMinute:(NSInteger)minimumExpectedMinute
+                                                   maximumExpectedHour:(NSInteger)maximumExpectedHour
+                                                 maximumExpectedMinute:(NSInteger)maximumExpectedMinute;
+
```

Comment:
The problem here is that if we directly use `NSDateComponents` objects on predicates, you cannot use the *greater than* and *less than* operators. You have to manually access their components to compare them. And I think `NSDateComponents` has too many fields to provide comprehensive support.

So, after discussing with John, we ended leaving this bare-bones method for the very basic (and hopefully common) use case.

Discussion in the [original PR](https://github.com/ResearchKit/ResearchKit/pull/151#issuecomment-102458319):

> [Me]: Yeah, there are too many `NSDateComponents` for covering all of them. But I did this because of a comment in `ORKTimeOfDayQuestionResult`'s *HeaderDoc*: `/** Typically only hour, minute, and AM/PM data are of interest. */`. Couldn't this method be useful anyway? If the developer needs more, she can just define further methods or manually create more predicates.

And we ended up deciding to keep it in an offline discussion.